### DISCUSSION TITLE
CSV Output

### DISCUSSION THREAD

**Discussion by Aaron668-5921** [ReagentX/imessage-exporter#403] [Ideas]
I've been looking through the discussions and documentation and, while I may have missed it, I'm not seeing an option for a structured text only output, CSV would be wonderful (or some other delimiter if preferred).  If I'm not missing an existing feature, could such be added?

**Comment by ReagentX** [comment]
I'm happy to work on this if there's enough interest, but we must think about what a feature-complete CSV export would look like. Can you provide some examples of how you would expect replies/threads, reactions, sticker types, edits, etc. to look? The full list of features is [here](https://github.com/ReagentX/imessage-exporter/blob/develop/docs/features.md).

**Reply by Aaron668-5921** [reply]
Happily. Ideally I’d like to build a full mockup in Excel, but I’m limited atm and want to reply in a more basic way now while I can, then if needed can come back and give a more full answer. // Msg_snd_time, msg_sent_by_Name, msg_content, is_edited_flag, msg_content_v0, msg_content_v1, msg_content_v2, msg_content_v3, msg_content_v4, msg_content_v5, msg_content_v6, msg_content_v7, tapback_flag, tapback_by_name, tapback_timestamp, tapback_content, is_reply_flag, replying_to_content, replying_to_sender, sent_by_Ph#, sent_by_emailI’m probably forgetting something(s) important, but does this help?

**Reply by moncubus** [reply]
Everything included in the TXT export would be useful to deal with in a structured form via CSV. Easier to use the data, filter by the sender, date, etc., in a spreadsheet.

I'd see each message as an individual row.

**Comment by westover** [comment]
Hi! I am definitely interested in a different type of export beyond just HTML and txt. Personally JSON I think makes sense because message objects could have no standard fields but CSV also makes sense. I built a whole parsing framework based on `bs4` in python. Frankly any structured format that is easier to parse makes sense to me, HTML is kinda messy for all of this. It would also be nice to have features be able to be toggled, like filter all app messages or something.

To put further context, my strategy and goal is to make these dumps more easy to share with conversational AI or databases.

**Reply by westover** [reply]
My background is in python, so I wrote my wrapper in python. Happy to share code and details of how I am doing that (haven't uploaded it to GitHub yet)

**Reply by ReagentX** [reply]
Related: #232. As of yet, none of the issues for new format types have garnered much attention nor has anyone provided a sample of what a [full-featured](https://github.com/ReagentX/imessage-exporter/blob/develop/docs/features.md) export should look like.

**Comment by jwr456** [comment]
I would like to see csv output.  it would be useful to be able to import to excel, power query, power bi, etc and look for words in messages by sender and date.  In this output format, I don't think it's as important to have all the features of html with pictures, stickers, and other rich media.  At least for me, its date, sender, message content in text.

**Reply by ReagentX** [reply]
If that's all you want, check out the docs for the [library](https://docs.rs/imessage-database/latest/imessage_database/)!
